app/models/journeyDomain/item/ItemDomain.scala

```diff
@@ -224,34 +213,34 @@ object ItemDomain {
     AddSupplyChainActorYesNoPage(itemIndex)
       .filterOptionalDependent(identity)(SupplyChainActorsDomain.userAnswersReader(itemIndex))
 
-  private def isConsignmentPreviousDocDefined(itemIndex: Index): UserAnswersReader[Option[DocumentsDomain]] =
-    DocumentsSection.arrayReader
-      .map {
-        _.validateAsListOf[Document]
-          .exists(
-            x => x.attachToAllItems && x.`type` == Previous
-          )
-      }
-      .flatMap {
-        case true  => AddDocumentsYesNoPage(itemIndex).filterOptionalDependent(identity)(DocumentsDomain.userAnswersReader(itemIndex))
-        case false => DocumentsDomain.userAnswersReader(itemIndex).map(Some(_))
-      }
-
   def documentsReader(itemIndex: Index): UserAnswersReader[Option[DocumentsDomain]] = {
 
-    val externalPages: UserAnswersReader[(DeclarationType, Boolean)] = for {
+    val externalPages: UserAnswersReader[(String, Boolean)] = for {
       consignmentDecType    <- TransitOperationDeclarationTypePage.reader
       isGBOfficeOfDeparture <- CustomsOfficeOfDeparturePage.reader.map(_.startsWith(GB))
     } yield (consignmentDecType, isGBOfficeOfDeparture)
 
+    def isConsignmentPreviousDocDefined(itemIndex: Index): UserAnswersReader[Option[DocumentsDomain]] =
```

Comment:
This may be more readable if you build it out as a for comp:

for {
      document <- DocumentsSection.arrayReader
      documents = document.validateAsListOf[Document]
      ...

Reply:
Ah with us being in an awkward position with the merges, this was actually part of Jack's branch (https://github.com/hmrc/ctc-departure-items-frontend/pull/193) - I merged his CTCP-3661 branch in to cater for the conflicts